core/CoreUtils.hpp

```diff
@@ -38,4 +46,14 @@ namespace olympia::coreutils
         return core_types::RegFile::RF_INVALID;
     }
 
+    inline core_types::RegFile determineRegisterFile(const std::string &target_name)
+    {
+        if (target_name == "alu" || target_name == "br") {
+            return core_types::RF_INTEGER;
+        } else if (target_name == "fpu") {
+            return core_types::RF_FLOAT;
+        }
+        sparta_assert(false, "Not supported this target: " << target_name);
+    }
```

Comment:
`gcc` sometimes hates functions with no returns -- and it's not smart enough to know that `sparta_assert` is an actual assert (yes, compilers actually optimize for that).   Suggest adding a bad return and/or `__builtin_unreachable();`

Reply:
Added a return of core_types::RF_INVALID